torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -242,9 +242,48 @@ Value* TracingState::getOutput(const IValue& iv, size_t i) {
         fmap(tuple, [&](const IValue& ival) { return getOutput(ival, i); }));
     graph->insertNode(tuple_node);
     return tuple_node->output();
+  } else if (iv.isGenericDict()) {
+    if (tracing_mode_strict) {
+      throw std::runtime_error(
+          "Encountering a dict at the output of the tracer" +
+          std::string(STRICT_TRACER_MSG));
+    }
+    auto dict = iv.toGenericDict();
+    TypePtr key_type = dict.keyType();
+    TypePtr value_type = dict.valueType();
+
+    if (!(key_type->cast<StringType>() || key_type->cast<TensorType>()) ||
+        !value_type->cast<TensorType>()) {
+      std::ostringstream os;
+      os << "output " << i << " (" << dict << ") of traced region "
+         << "cannot be understood by the tracer, only dict[str, Tensor] "
+         << "or dict[Tensor, Tensor] can be a dictionary output of a traced function";
+      throw std::runtime_error(os.str());
+    }
+    const auto order = iterationOrder(dict);
```

Comment:
Not an issue with your PR, but I looked at the declaration of `iterationOrder` and I'm not sure what it does :p. It could use some comments or a rename to indicate its function

Reply:
Can you (or someone) add that as a comment?

Reply:
did this in https://github.com/pytorch/pytorch/pull/36609